auslib/blobs/systemaddons.py

```diff
@@ -0,0 +1,92 @@
+from auslib.AUS import isForbiddenUrl
+from auslib.blobs.base import Blob
+from auslib.errors import BadDataError
+
+
+class SystemAddonsBlob(Blob):
+    jsonschema = "systemaddons.yml"
+
+    def __init__(self, **kwargs):
+        Blob.__init__(self, **kwargs)
+        if "schema_version" not in self:
+            self["schema_version"] = 5000
+
+    def getAddonsForPlatform(self, platform):
+        for v in self["addons"]:
+            if platform in self["addons"][v]["platforms"] or "default" in self["addons"][v]["platforms"]:
+                yield v
+
+    def getResolvedPlatform(self, addon, platform):
+        if platform in self['addons'][addon]['platforms']:
+            return self['addons'][addon]['platforms'][platform].get('alias', platform)
+        if "default" in self['addons'][addon]['platforms']:
+            return "default"
+        raise BadDataError("No platform '%s' or default in addon '%s'",
+                           platform, addon)
+
+    def getPlatformData(self, addon, platform):
+        platform = self.getResolvedPlatform(addon, platform)
+        try:
+            return self['addons'][addon]['platforms'][platform]
+        except KeyError:
+            raise BadDataError("No platform '%s' in addon '%s'", platform,
+                               addon)
+
+    def shouldServeUpdate(self, updateQuery):
+        # SystemAddon updates should always be returned. It is the responsibility
+        # of the client to decide whether or not any action needs to be taken,
+        # similar to GMP
+        return True
+
+    # If there are are no updates, we have a special response for SystemAddons
+    # blobs. We return <updates></updates>, without the addons tags.
+    def noUpdates(self, updateQuery, whitelistedDomains):
```

Comment:
I think this would be simpler to read and implement if you reversed it and wrote a "hasUpdates" function instead. It would mean that the conditions in the header/footer methods below would not need a double negative ("if self.hasUpdates(...)" instead of "if not self.noUpdates").